src/components/Wallet/AccountSelector/AccountSelector.tsx

```diff
@@ -0,0 +1,119 @@
+import { MutableRefObject, useMemo } from 'react';
+import { Account } from '../../../generated/graphql';
+import { AccountItem } from './AccountItem/AccountItem';
+import { Button, ButtonKind } from '../../Button/Button';
+import './AccountSelector.scss';
+
+export interface AccountSelectorProps {
+  accounts?: Account[];
+  account?: Account;
+  onAccountSelected: (asset: Account) => void;
+  innerRef: MutableRefObject<HTMLDivElement | null>;
+  closeModal: () => void;
+  setActiveAccount: Function;
+  isExtensionAvailable: boolean;
+}
+
+/**
+ * Renders a list of accounts that the user can select an account from
+ */
+export const AccountSelector = ({
+  accounts,
+  onAccountSelected,
+  account,
+  innerRef,
+  closeModal,
+  setActiveAccount,
+  isExtensionAvailable,
+}: AccountSelectorProps) => {
+  const activeAccount = useMemo(() => account, [account]);
+
+  return (
+    <div className="account-selector" ref={innerRef}>
+      <div className="account-selector__content-wrapper">
+        <div className="d-flex flex-align-space mx-3 my-3 account-selector__heading">
+          <div>
+            {isExtensionAvailable ? 'Select an account' : 'Install extension'}
+          </div>
+          <div
+            className="account-selector__close-modal-btn"
+            onClick={() => closeModal()}
+          >
+            x
+          </div>
+        </div>
+        {isExtensionAvailable ? (
+          <>
+            {accounts?.length ? (
+              <div className="account-selector__accounts-list">
+                {accounts?.map((account, i) => (
+                  <AccountItem
+                    key={i}
+                    onClick={() => onAccountSelected(account)}
```

Comment:
this should be a `useCallback`

Reply:
`onAccountSelected` is now implemented with `useCallback` and passed down from `useModalPortalElement.tsx`